### PR TITLE
(Issue #28) Added an optional validation set argument to the trainer

### LINKED ISSUE #28: add validation option on trainer

### PR DESCRIPTION
**Pull request by stefanos-vlachos**
@ChrisNick92 @ellievak 

# Summary
This PR introduces an optional validation set argument to the training loop.

# Additions

- The **validation_dset** argument was added to [Trainer](https://github.com/magcil/deepaudio-x/blob/28-add-validation-option-on-trainer/src/deepaudiox/loops/trainer.py). This argument defaults to **None**.
- The **_setup_dataloaders()** function of  [Trainer](https://github.com/magcil/deepaudio-x/blob/28-add-validation-option-on-trainer/src/deepaudiox/loops/trainer.py) was modified to also handle validation set; If **validation_dset** is provided by the user, it is loaded into a PyTorch DataLoader, otherwise, the provided **train_dset** is split into train and validation.

# Checklist
- [x] Scanned and formatted the commited code using ruff.

### CODE REVIEW

**Review by ChrisNick92 — APPROVED** [review]
Ok the warnings are now gone. I merge this.
